src/main/kotlin/app/utils/FileHelper.kt

```diff
@@ -3,12 +3,45 @@
 
 package app.utils
 
+import java.io.File
+import java.net.URLDecoder
+import java.nio.file.Files
 import java.nio.file.Paths
+import java.nio.file.Path
 
 object FileHelper {
+    private val dirName = "data"
+    private val jarPath = getJarPath()
+    private val settingsPath = jarPath.resolve(dirName)
+
+    fun getPath(name: String, vararg parts: String): Path {
+        val path = settingsPath.resolve(Paths.get("", *parts))
+        if (Files.notExists(path)) {
+            Files.createDirectories(path)
+        }
+        return path.resolve(name)
+    }
+
+    fun getFile(name: String, vararg parts: String): File {
+        return getPath(name, *parts).toFile()
+    }
+
+    fun isNotExists(name:String, vararg parts: String): Boolean {
+        return Files.notExists(getPath(name, *parts))
+    }
+
     fun getFileExtension(path: String): String {
         val fileName = Paths.get(path).fileName.toString()
-        return fileName.substringAfterLast(
-                delimiter = '.', missingDelimiterValue = "")
+        return fileName.substringAfterLast(delimiter = '.',
+                                           missingDelimiterValue = "")
+    }
+
+    fun getJarPath(): Path {
+        val fullPathString = URLDecoder.decode(FileHelper::class.java
+            .protectionDomain.codeSource.location.toURI().path, "UTF-8")
+        val fullPath = Paths.get(fullPathString)
+        val root = fullPath.root
+        // Removing jar filename.
+        return root.resolve(fullPath.subpath(0, fullPath.nameCount - 1))
```

Comment:
might be unexpected behavior for the user, if the user moved a jar file, and we put all our stuff next to it, @sergey-surkov is it ok?

Reply:
Why would the user want to do it? We use bash script that points to a specific location of the JAR to run sourcerer, all will be broken anyway.